### PR TITLE
fix(sdk): sort params to fix flaky tests.

### LINKED ISSUE #661: pipelineloop test will occasionally fail because the generated parameters are not in order
Also I realize one of the pipelineloop test will occasionally fail because the generated parameters are not in order. FYI https://github.com/kubeflow/kfp-tekton/pull/512/checks?check_run_id=3004430660

Let me open another issue for this.

_Originally posted by @Tomcli in https://github.com/kubeflow/kfp-tekton/issues/660#issuecomment-876033213_

### PR DESCRIPTION
**Pull request by ScrapCodes**
**Which issue is resolved by this Pull Request:** 
Resolves #661

**Description of your changes:**
Sorted params in `PipelineRun.Spec.Params`.

**Environment tested:**

* Python Version (use `python --version`):
* Tekton Version (use `tkn version`):
* Kubernetes Version (use `kubectl version`):
* OS (e.g. from `/etc/os-release`):

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention). 

- [ ] Do you want this pull request (PR) cherry-picked into the current release branch?

    [Learn more about cherry-picking updates into the release branch](https://github.com/kubeflow/pipelines/blob/master/RELEASE.md#cherry-picking-pull-requests-to-release-branch).

### CODE REVIEW

File: sdk/python/kfp_tekton/compiler/compiler.py [hunk 1]
```diff
@@ -1352,7 +1352,9 @@ def _create_and_write_workflow(self,
                                                          pipeline_loop_crs, recursive_tasks_names)
         inlined_as_taskSpec.extend(e)
         workflow['spec']['pipelineSpec']['tasks'] = workflow_tasks
-
+        # Preserve order of params, required by tests.
+        if 'params' in workflow['spec']:
+          workflow['spec']['params'] = sorted(workflow['spec']['params'], key=lambda kv: (kv['name']))
```
Comment (pugangxa) [inline sdk/python/kfp_tekton/compiler/compiler.py:1357]: So we already have this logic but just for step 1 of inline task. This is for after step 2.

Comment (ScrapCodes) [reply sdk/python/kfp_tekton/compiler/compiler.py:1357]: Thanks @pugangxa , That one sorts inside a CR `pipeline_loop_crs[i]['spec']['pipelineSpec']['params']` and this one sorts `PipelineRun.Spec.Params`.

**Pull request merged**